### PR TITLE
Added Authentication

### PR DESCRIPTION
**Pull request by SaurabhChawda**
Added Authentication and Routing in Quiz App.
Screenshot (358)
Screenshot (361)
Screenshot (360)
Screenshot (359)

### CODE REVIEW

**Review by ultronjs — APPROVED** [review]
The code look well structured and clean
Functionality also working fine
Good to merge

File: src/Context/AuthContext.jsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      console.log(error);
+      toast.error(error);
+    }
+  };
+
+  //User Logout
+
+  const logoutCredentials = () => {
+    try {
+      localStorage?.removeItem("login");
```
Comment (Namrata-J) [inline src/Context/AuthContext.jsx:56]: liked the way you used optional chaining here!

**Review by Namrata-J — APPROVED** [review]
The rest of the code looks good to me.
Great work on UI, it looks clean.
You can merge it.

**Comment by SaurabhChawda** [comment]
> The code look well structured and clean Functionality also working fine Good to merge

ThankYou

**Comment by SaurabhChawda** [comment]
> The rest of the code looks good to me. Great work on UI, it looks clean. You can merge it.

ThankYou

**Pull request merged**